src/meta/processors/admin/AdminClient.cpp

```diff
@@ -335,6 +335,50 @@ folly::Future<Status> AdminClient::checkPeers(GraphSpaceID spaceId, PartitionID
   return fut;
 }
 
+folly::Future<nebula::cpp2::ErrorCode> AdminClient::clearSpace(GraphSpaceID spaceId,
+                                                               const std::vector<HostAddr>& hosts) {
+  folly::Promise<nebula::cpp2::ErrorCode> promise;
+  auto f = promise.getFuture();
+
+  std::vector<folly::Future<StatusOr<nebula::cpp2::ErrorCode>>> futures;
+  for (auto& host : hosts) {
+    folly::Promise<StatusOr<nebula::cpp2::ErrorCode>> pro;
+    futures.emplace_back(pro.getFuture());
+
+    storage::cpp2::ClearSpaceReq req;
+    req.space_id_ref() = spaceId;
+    getResponseFromHost(
+        Utils::getAdminAddrFromStoreAddr(host),
+        std::move(req),
+        [](auto client, auto request) { return client->future_clearSpace(request); },
+        [](auto&& resp) -> nebula::cpp2::ErrorCode { return resp.get_code(); },
+        std::move(pro));
+  }
+
+  folly::collectAll(std::move(futures))
```

Comment:
prefer to return this future directly rather than using promise.

Reply:
In order to be consistent with other interfaces. And the upper layer also needs a future as return value.